app/code/Magento/CatalogUrlRewrite/Plugin/Catalog/Model/Product/ProductToWebsiteChange.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogUrlRewrite\Plugin\Catalog\Model\Product;
+
+use Magento\Catalog\Model\Product\Action as ProductAction;
+use Magento\Catalog\Model\ResourceModel\Product\Collection;
+use Magento\CatalogUrlRewrite\Model\Product\AppendUrlRewritesToProducts;
+use Magento\CatalogUrlRewrite\Model\ProductUrlRewriteGenerator;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\RequestInterface;
+use Magento\Store\Api\StoreWebsiteRelationInterface;
+use Magento\UrlRewrite\Model\UrlPersistInterface;
+use Magento\UrlRewrite\Service\V1\Data\UrlRewrite;
+
+/**
+ * Update URL rewrites after website change
+ */
+class ProductToWebsiteChange
```

Comment:
Plugin name is confusing. Please change it to url rewrite related.

Reply:
renamed